### DISCUSSION TITLE
Rename this repository, adobe/spectrum-tokens, to adobe/spectrum-design-data

### DISCUSSION THREAD

**Discussion by GarthDB** [adobe/spectrum-design-data#361] [Ideas]
# Problem statement

In this repository, `adobe/spectrum-tokens`, we are adding more than just token data.

Specifically, we are currently adding JSON schema for component options and have plans to add support for capturing component anatomy data in the future.

# History

Historically, the token data and management tools were part of a repository named `spectrum-dna`. It has been spectrum-tokens for a few years. We have since restructured the repository as a monorepo to manage several integrated packages, including token data manipulation tools. We are currently building a token diff report generating tool and moving the component options schema into their own package published on NPM.

# Proposal

Change the name from `adobe/spectrum-tokens` to `adobe/spectrum-design-data`.

# Alternatives considered

I considered changing it back to `spectrum-dna` because I liked the name and historical context, but spectrum-design-data is the most descriptive of the purpose of the packages and data in this project.

# Consequences or Required Actions

Because I am not proposing changing the name of the [@adobe/spectrum-tokens](https://www.npmjs.com/package/@adobe/spectrum-tokens) NPM package, most projects will not be impacted by this change.

It will impact the website for the token visualizer, but we could create a way to redirect to the new URL.

It will also impact the GitHub actions that keep the [adobe/spectrum-tokens-studio-data](https://github.com/adobe/spectrum-tokens-studio-data) repository in sync with this repository.

**Comment by jnjosh** [comment]
I think we do struggle a lot overloading the term `tokens` in our ecosystem so I welcome this change minimally being more specific and giving the term "design data" a boost. `Tokens` become a sort of conceptual "output" of design data not the entirety of it. So… yes I'm agreeing that this is good. Let's agree to agree.

**Comment by GarthDB** [comment]
I'm doing more work on this stuff, here's the [pr](https://github.com/adobe/spectrum-tokens/pull/558) for it.

**Comment by GarthDB** [comment]
The rename is done and I'm adding redirects for the static pages like https://opensource.adobe.com/spectrum-tokens/s2-tokens-viewer/ currently.

**Comment by GarthDB** [comment]
## 🚨 Breaking Changes Update - Need Feedback

The repository rename is now complete in PR #632, but we've identified **breaking changes** that affect consumers of our published packages. We need confirmation that downstream consumers are aware and can accommodate these changes.

### What Changed

**1. JSON Schema `$id` URIs**

All schema canonical identifiers changed from:
```
https://opensource.adobe.com/spectrum-tokens/schemas/...
```
to:
```
https://opensource.adobe.com/spectrum-design-data/schemas/...
```

**2. Internal Schema `$ref` Paths**

Changed from absolute URLs to relative paths:
- Old: `"$ref": "https://opensource.adobe.com/spectrum-tokens/schemas/token-types/alias.json"`
- New: `"$ref": "alias.json"` (with `$id` as base URI)

### Breaking Change Impact

**Affected Packages:**
- `@adobe/spectrum-tokens` → Major version bump
- `@adobe/spectrum-component-api-schemas` → Major version bump

**Who is affected:**

Any external tools, validators, or schemas that reference our schemas by their `$id` URL will need to update their references.

**What's NOT breaking:**
- ✅ NPM package names (unchanged)
- ✅ Git operations (GitHub redirects automatically)
- ✅ GitHub Pages URLs (we've set up HTTP redirects)

### ⚠️ Action Required

**We need feedback from package consumers:**

If you consume `@adobe/spectrum-tokens` or `@adobe/spectrum-component-api-schemas`:
1. Do any of your tools reference our JSON schemas by their `$id` URI?
2. Are you comfortable with this being a major version bump?
  - @adobe/spectrum-tokens: 13.16.0 → 14.0.0
  - @adobe/spectrum-component-api-schemas: 5.0.1  → 6.0.0
  - @adobe/spectrum-design-data-mcp: 1.0.11 → 1.0.12
3. Do you need any migration guidance or tooling support?

### Additional Impact: Published Dist Files

Important clarification: The `$schema` URL changes affect **both source and dist files** in the published `@adobe/spectrum-tokens` package.

**Scope of Impact:**
- **Source files** (`src/*.json`): All `$schema` properties updated
- **Dist files** (`dist/json/variables.json`, `dist/json/drover.json`): **234+ instances** of `$schema` properties with the new URLs

**Example change in published dist files:**
```json
{
  "white": {
    "$schema": "https://opensource.adobe.com/spectrum-design-data/schemas/token-types/color.json",
    "value": "rgb(255, 255, 255)",
    ...
  }
}
```

**Additional consumers who may be affected:**
- Tools that validate token data using embedded `$schema` properties
- Parsers that extract or cache `$schema` URLs from dist files
- Any automation that relies on exact schema URL matches

The Style Dictionary build process copies these `$schema` properties from source to dist, so the URL changes propagate through to the published package output.

### Redirects in Place

For reference, we've set up the following in the old `adobe/spectrum-tokens` repository:
- Main site → https://opensource.adobe.com/spectrum-design-data/
- Visualizer → https://opensource.adobe.com/spectrum-design-data/visualizer/
- S2 Visualizer → https://opensource.adobe.com/spectrum-design-data/s2-visualizer/
- S2 Tokens Viewer → https://opensource.adobe.com/spectrum-design-data/s2-tokens-viewer/
- Release Timeline → https://opensource.adobe.com/spectrum-design-data/release-timeline/
- Catch-all 404 → Redirects to new paths

Related: #632

**Comment by GarthDB** [comment]
## ✅ Repository Rename Complete

**Discussion opened:** June 20, 2024  
**Completed:** November 2025

### What Was Accomplished

The repository has been successfully renamed from `adobe/spectrum-tokens` to `adobe/spectrum-design-data` to better reflect that this monorepo contains more than just design tokens (component schemas, diff tools, MCP server, etc.).

### Released Versions

| Package | Version | Change Type |
|---------|---------|-------------|
| `@adobe/spectrum-tokens` | 14.0.0 | **Major** (breaking) |
| `@adobe/spectrum-component-api-schemas` | 6.0.0 | **Major** (breaking) |
| `@adobe/spectrum-design-data-mcp` | 1.0.12 | Patch |
| `@adobe/token-diff-generator` | 2.5.2 | Patch |
| `@adobe/spectrum-component-diff-generator` | 1.4.1 | Patch |

### What Changed

**Breaking Changes:**
- JSON Schema `$id` URIs changed from `spectrum-tokens` → `spectrum-design-data`
- Internal schema `$ref` paths simplified to relative references
- All `$schema` properties in source and dist files updated (234+ instances)

**Not Breaking:**
- ✅ NPM package names unchanged
- ✅ GitHub automatically redirects `adobe/spectrum-tokens` → `adobe/spectrum-design-data`

### Redirects Configured

The following redirects are in place from the old `spectrum-tokens` URLs:

| Old URL | New URL |
|---------|---------|
| `opensource.adobe.com/spectrum-tokens/` | `opensource.adobe.com/spectrum-design-data/` |
| `.../spectrum-tokens/visualizer/` | `.../spectrum-design-data/visualizer/` |
| `.../spectrum-tokens/s2-visualizer/` | `.../spectrum-design-data/s2-visualizer/` |
| `.../spectrum-tokens/s2-tokens-viewer/` | `.../spectrum-design-data/s2-tokens-viewer/` |
| `.../spectrum-tokens/release-timeline/` | `.../spectrum-design-data/release-timeline/` |
| `.../spectrum-tokens/schemas/*` | `.../spectrum-design-data/schemas/*` |

### Verification

- ✅ All NPM packages published successfully
- ✅ GitHub Pages schemas accessible at new URLs
- ✅ GitHub repository redirect working
- ✅ All CI checks passing

### Related PRs

- #632 - Main implementation
- #638 - URL updates  
- #641 - Version release

---

🎉 **Closing this discussion as complete.** Thanks to everyone who provided feedback!
